main.py

```diff
@@ -134,11 +140,29 @@ def __init__(self):
         self.isBallSelected = False             # Boolean value if a ball is selected 
         self.selectedBall = Ball(0)             # Variable to hold the selected ball object
         self.balls = []                         # List that holds all ball objects in COORDS
+        self.buttons = []                       # List that holds all button objects
+        self.moves = []                         # List that holds the last states to undo
 
         # Append balls[]-list with a new ball for each ball in COORDS
         for coord in COORDS:
             self.balls.append(Ball(len(self.balls)))
 
+        # Add Restart and Undo Button
+        self.buttons.append(Button((120,10,120,40), "Restart", self.__init__))
+        self.buttons.append(Button((260,10,120,40), " Undo", self.undo))
+
+        # Add initial state to moves
+        self.moves.append(START)
+
+    def undo(self):
```

Comment:
Kann man diskutieren, ob das eine Funktion der Klasse "Game" oder eines speziellen "Undo-Buttons" sein sollte.
Mögliche Redesign Entscheidung im späteren Verlauf.

Reply:
Ist eine gute Frage, der Einfachheit halber würde ich das aber erstmal in der "Game" Klasse lassen, damit die Button Klasse generisch bleibt und nicht noch irgendwelche Abhängigkeiten (states, moves, etc.) braucht